notebooks/06 Modelling - classification.ipynb

```diff
@@ -0,0 +1,993 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Modelling - Classification\n",
+    "\n",
+    "This notebook explores classification models to **predict risk of becoming a long stayer** as a baseline to the [Long Stayer Risk Stratification](https://github.com/nhsx/skunkworks-long-stayer-risk-stratification) model.\n",
+    "\n",
+    "This notebook is broken down into:\n",
+    "\n",
+    "1. Converting the length of stay into a relative risk\n",
+    "1. Training a range of baseline models using cross validation\n",
+    "3. Testing final models on a test dataset\n",
+    "4. Exploring in more detail the best performing baseline model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653643592936
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import math\n",
+    "import pickle\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import plotly.express as px\n",
+    "from catboost import CatBoostClassifier\n",
+    "from sklearn import preprocessing\n",
+    "from sklearn.dummy import DummyClassifier\n",
+    "from sklearn.ensemble import RandomForestClassifier\n",
+    "from sklearn.linear_model import LogisticRegression\n",
+    "from sklearn.metrics import balanced_accuracy_score, f1_score, roc_auc_score\n",
+    "from sklearn.model_selection import GridSearchCV, cross_validate, train_test_split\n",
+    "from sklearn.tree import DecisionTreeClassifier\n",
+    "from sklearn.utils.class_weight import compute_class_weight\n",
+    "from xgboost import XGBClassifier\n",
+    "\n",
+    "pd.set_option(\"display.max_rows\", None)\n",
+    "pd.set_option(\"display.max_columns\", None)\n",
+    "\n",
+    "%matplotlib inline\n",
+    "plt.rcParams[\"figure.figsize\"] = [15, 8]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653643561513
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Helper functions\n",
+    "\n",
+    "\n",
+    "def train_model(gsc, X_train, y_train):\n",
```

Comment:
I would add this function to a py file and add an input for metrics. This then can be imported in to both notebooks and helps with code maintainability.